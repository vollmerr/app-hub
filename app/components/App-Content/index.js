
import styled from 'styled-components';

import theme from 'utils/theme';

const AppContent = styled.div`
  display: flex;
  flex: 1;
  padding: 0 15px;
  background-color: ${theme.neutralLighterAlt};
  height: 100vh;
  overflow: auto;
`;

export default AppContent;
