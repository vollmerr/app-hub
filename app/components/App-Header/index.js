/**
*
* AppHeader
*
*/

import React from 'react';
// import styled from 'styled-components';

class AppHeader extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <div>
      </div>
    );
  }
}

AppHeader.propTypes = {

};

export default AppHeader;
