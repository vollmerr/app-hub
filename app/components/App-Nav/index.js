/**
*
* AppPanel
*
*/

import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Nav } from 'office-ui-fabric-react/lib/Nav';

import theme from 'utils/theme';

const NavItems = styled(Nav) `
  background: ${theme.neutralLighterAlt};
  li > div {
    background: ${theme.neutralLighterAlt};

    a {
      text-decoration: none;
      display: block;

      &: hover,
      &: focus,
      &: active {
        background: ${theme.neutralLight};
      }
    }

    &:hover a,
    &.is-selected a {
      background: ${theme.neutralLight};
    }
  }
`;


class AppNav extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { routes, onClick } = this.props;

    return (
      <NavItems
        groups={[{ links: routes }]}
        onRenderLink={(link) => (
          link.path ?
            <Link to={link.path} key={link.key} onClick={onClick}>{link.name}</Link> :
            <a href={link.href} key={link.key}>{link.name}</a>
        )}
      />
    );
  }
}

AppNav.propTypes = {

};

export default AppNav;
