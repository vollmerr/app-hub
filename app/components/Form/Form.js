import styled from 'styled-components';

import theme from 'utils/theme';


const Form = styled.form`
  min-height: calc(${(props) => props.vh}vh - ${theme.hub.headerHeight} - 30px - ${(props) => props.margin});
  padding: 15px;
  margin: 15px 0;
  background: ${theme.white};
`;

Form.defaultProps = {
  noValidate: true,
  vh: 100,
  margin: 0,
};

export default Form;
