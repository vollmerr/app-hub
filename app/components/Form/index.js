export { default as Field } from './Field';
export { default as FieldChecks } from './FieldChecks';
export { default as FieldDate } from './FieldDate';
export { default as FieldFile } from './FieldFile';
export { default as FieldRadios } from './FieldRadios';
export { default as FieldSelect } from './FieldSelect';
export { default as FieldText } from './FieldText';
export { default as FieldToggle } from './FieldToggle';
export { default as FormButtons } from './FormButtons';
export { default as FormSection } from './FormSection';
