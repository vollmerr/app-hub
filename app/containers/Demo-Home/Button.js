import React from 'react';

const Button = ({ onClick, disabled, text }) => ( // eslint-disable-line
  <button
    onClick={onClick}
    disabled={disabled}
    style={{ background: disabled ? '#666' : '#333', color: '#fff', padding: '10px 15px', cursor: 'pointer' }}
  >{text}</button>
);

export default Button;
