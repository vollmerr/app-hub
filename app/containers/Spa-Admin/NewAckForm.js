import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Form } from 'react-final-form';

import appPage from 'containers/App-Container/appPage';
import { StyledForm, FormButtons } from 'components/Form';
import theme from 'utils/theme';

import validate from './validate';


export const Fields = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
`;

// min-width = xs breakpoint - 2 * 15px margin (Content) - 2 * 15px padding (Form) - 15px margin (Form)
export const FieldSection = styled.div`
  flex: 100%;
  padding-right:15px;

  @media (min-width: ${theme.breakpoints.md}px) {
    flex: 50%;
  }
`;


function mapSection(fields, section) {
  return (
    <FieldSection>
      {
        section.map((name) => {
          const { component: C, ...props } = fields[name];

          return (
            <C {...props} key={name} />
          );
        })
      }
    </FieldSection>
  );
}


/**
 * Form for new acknowledgments, available only to admins
 */
export class NewAckForm extends React.PureComponent {
  renderForm = ({
    reset,
    pristine,
    submitting,
    handleSubmit,
  }) => {
    const {
      title,
      fields,
      sections,
    } = this.props;

    const buttonProps = {
      reset,
      disabled: pristine || submitting,
    };

    return (
      <StyledForm onSubmit={handleSubmit} margin={theme.app.subNavHeight}>
        <h3>{title}</h3>

        <Fields>
          {mapSection(fields, sections.left)}
          {mapSection(fields, sections.right)}
        </Fields>

        <FormButtons {...buttonProps} />
      </StyledForm>
    );
  }

  render() {
    const { onSubmit } = this.props;

    const formProps = {
      onSubmit,
      validate,
      render: this.renderForm,
    };

    return <Form {...formProps} />;
  }
}


const { func, string, object } = PropTypes;

NewAckForm.propTypes = {
  title: string.isRequired,
  fields: object.isRequired,
  sections: object.isRequired,
  onSubmit: func.isRequired,
};

export default appPage(NewAckForm);
