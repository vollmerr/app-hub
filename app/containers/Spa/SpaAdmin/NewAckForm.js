import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Form } from 'react-final-form';

import { FormSection, FormButtons } from '../../../components/Form';
import theme from '../../../utils/theme';

import validate from './validate';


export const Fields = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
`;


// min-width = xs breakpoint - 2 * 15px margin (Content) - 2 * 15px padding (Form) - 15px margin (Form)
export const FieldSection = styled.div`
  flex: 100%;
  padding-right:15px;
  @media (min-width: ${theme.breakpoints.md}px) {
    flex: 50%;
  }
`;


function mapSection(fields, section) {
  return (
    <FieldSection>
      {
        section.map((name) => {
          const { component: C, ...props } = fields[name];

          return (
            <C {...props} key={name} />
          );
        })
      }
    </FieldSection>
  );
}


const style = {
  padding: theme.hub.padding + theme.app.commandBarHeight,
};


/**
 * Form for new acknowledgments, available only to admins
 */
export class NewAckForm extends React.PureComponent {
  renderForm = ({
    reset,
    pristine,
    submitting,
    handleSubmit,
  }) => {
    const {
      title,
      fields,
      sections,
    } = this.props;

    const formProps = {
      onSubmit: handleSubmit,
      ...style,
    };

    const buttonProps = {
      reset,
      disabled: pristine || submitting,
    };

    return (
      <FormSection {...formProps}>
        <h3>{title}</h3>

        <Fields>
          {mapSection(fields, sections.left)}
          {mapSection(fields, sections.right)}
        </Fields>

        <FormButtons {...buttonProps} />
      </FormSection>
    );
  }

  render() {
    const { onSubmit } = this.props;

    const formProps = {
      onSubmit,
      validate,
      render: this.renderForm,
    };

    return <Form {...formProps} />;
  }
}


const { func, string, object } = PropTypes;

NewAckForm.propTypes = {
  title: string.isRequired,
  fields: object.isRequired,
  sections: object.isRequired,
  onSubmit: func.isRequired,
};


export default NewAckForm;
