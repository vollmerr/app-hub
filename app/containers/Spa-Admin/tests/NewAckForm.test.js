import React from 'react';
import { shallow } from 'enzyme';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

import { Form, FormButtons } from 'components/Form';
import spaFields, { newAckForm } from 'containers/Spa/fields';

import { NewAckForm } from '../NewAckForm';

const props = {
  title: 'test title',
  fields: spaFields,
  sections: newAckForm.sections,
  handleSubmit: jest.fn(),
  reset: jest.fn(),
  pristine: false,
  submitting: false,
  onSubmit: jest.fn(),
};


describe('<NewAckForm />', () => {
  let wrapper;
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<NewAckForm {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a `Form`', () => {
    expect(wrapper.find(Form).length).toEqual(1);
  });

  it('should render a title (h3)', () => {
    expect(wrapper.find(Form).find('h3').length).toEqual(1);
    expect(wrapper.find('h3').text()).toEqual(props.title);
  });

  it('should render a `FormButtons` section for action buttons', () => {
    expect(wrapper.find(Form).find(FormButtons).length).toEqual(1);
  });

  it('should submit and clear `DefaultButton`s in the buttons seciton', () => {
    expect(wrapper.find(FormButtons).find(DefaultButton).length).toEqual(2);
  });

  it('should render all the fields', () => {
    const len = newAckForm.sections.left.length + newAckForm.sections.right.length;
    expect(wrapper.find('WrappedField').length).toEqual(len);
  });

  it('should disable the buttons when `pristine` or `submitting`', () => {
    wrapper.setProps({ pristine: true });
    expect(wrapper.find(DefaultButton).at(0).prop('disabled')).toEqual(true);
    wrapper.setProps({ pristine: false, submitting: true });
    expect(wrapper.find(DefaultButton).at(1).prop('disabled')).toEqual(true);
  });

  it('should call onSubmit with redux-form`s handleSubmit when submitting the form', () => {
    wrapper.find(Form).simulate('submit');
    expect(props.handleSubmit).toHaveBeenCalledWith(props.onSubmit);
  });
});
