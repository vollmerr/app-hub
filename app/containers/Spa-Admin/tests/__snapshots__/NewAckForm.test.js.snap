// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewAckForm /> should render correctly 1`] = `
<Form__Form
  margin="40px"
  noValidate={true}
  vh={100}
>
  <h3>
    test title
  </h3>
  <NewAckForm__Fields>
    <NewAckForm__FieldSection>
      <WrappedField
        ariaLabel="Title of the acknowledgment"
        disabled={false}
        key="title"
        label="Title"
        name="title"
        placeholder="Enter acknowledgment title"
        required={true}
        validate={Array []}
      />
      <WrappedField
        ariaLabel="Date the acknowledgment takes effect"
        disabled={false}
        key="activeStartDate"
        label="Start Date"
        name="activeStartDate"
        placeholder="Select the start date"
        required={true}
        validate={
          Array [
            [Function],
          ]
        }
      />
      <WrappedField
        ariaLabel="Date the acknowledgment expires"
        disabled={false}
        key="activeEndDate"
        label="End Date"
        name="activeEndDate"
        placeholder="Select the end date"
        required={true}
        validate={
          Array [
            [Function],
          ]
        }
      />
      <WrappedField
        ariaLabel="AD groups that must acknowledge the policy"
        disabled={false}
        key="targetADGroup"
        label="Target Group(s)"
        name="targetADGroup"
        options={Array []}
        placeholder=""
        required={true}
        validate={Array []}
      />
    </NewAckForm__FieldSection>
    <NewAckForm__FieldSection>
      <WrappedField
        ariaLabel="Statement that will be placed under the details of the acknowledgment"
        disabled={false}
        key="ackStatement"
        label="Statement"
        name="ackStatement"
        placeholder="Enter the acknowledgment statement"
        required={true}
        validate={Array []}
      />
      <WrappedField
        ariaLabel="Details of the acknowledgment"
        disabled={false}
        key="detailsText"
        label="Details"
        multiline={true}
        name="detailsText"
        placeholder="Enter acknowledgment details"
        required={true}
        validate={Array []}
      />
      <WrappedField
        ariaLabel="Name of the file that must be read before acknowledging"
        disabled={false}
        key="fileName"
        label="Attachment Name"
        name="fileName"
        placeholder="Enter attachment name"
        required={false}
        validate={Array []}
      />
      <WrappedField
        ariaLabel="Attach the file that must be read before acknowledging"
        disabled={false}
        key="fileContent"
        label="Attachment"
        name="fileContent"
        placeholder="Attach a File"
        required={false}
        validate={Array []}
      />
    </NewAckForm__FieldSection>
  </NewAckForm__Fields>
  <FormButtons__Buttons>
    <CustomizedDefaultButton
      disabled={false}
      primary={true}
      text="Submit"
      type="submit"
    />
    <CustomizedDefaultButton
      disabled={false}
      onClick={[Function]}
      text="Clear"
    />
  </FormButtons__Buttons>
</Form__Form>
`;
